winpr/libwinpr/file/generic.c

```diff
@@ -801,58 +846,81 @@ HANDLE FindFirstFileA(LPCSTR lpFileName, LPWIN32_FIND_DATAA lpFindFileData)
 		return INVALID_HANDLE_VALUE;
 	}
 
-	/* Separate lpFileName into path and pattern components */
-	p = strrchr(lpFileName, '/');
-
-	if (!p)
-		p = strrchr(lpFileName, '\\');
+	oldErrno = errno;
```

Comment:
Why do we preserve errno here?

Reply:
right, could remove the restore and simply set to `0`